repository 에 type 없는 경우 만들기

현재 명시적으로 any를 사용한 부분이 110곳 가량 확인되었습니다.

<details><summary>Details</summary>

```
/home/scarf/repo/lib42/backend/backend/src/auth/auth.controller.ts
   27:42  error  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
   38:23  error  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
   49:19  error  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
   67:32  error  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
   79:19  error  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  110:19  error  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  145:46  error  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  176:19  error  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any

/home/scarf/repo/lib42/backend/backend/src/auth/auth.service.ts
  5:1  error  Definition for rule 'import/prefer-default-export' was not found  import/prefer-default-export

/home/scarf/repo/lib42/backend/backend/src/auth/auth.strategy.ts
  12:17  error  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  12:36  error  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  12:50  error  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  28:17  error  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  28:36  error  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  28:50  error  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  48:27  error  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any

/home/scarf/repo/lib42/backend/backend/src/auth/auth.validate.ts
  25:35  error  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  37:19  error  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any

/home/scarf/repo/lib42/backend/backend/src/auth/auth.validateDefaultNullUser.ts
  26:37  error  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  38:21  error  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any

/home/scarf/repo/lib42/backend/backend/src/book-info-reviews/controller/utils/errorCheck.ts
  13:20  error  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any

/home/scarf/repo/lib42/backend/backend/src/book-info-reviews/controller/utils/parseCheck.ts
   2:15  error  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
   7:20  error  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  14:10  error  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any

/home/scarf/repo/lib42/backend/backend/src/books/books.controller.spec.ts
  12:16  error  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  17:16  error  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any

/home/scarf/repo/lib42/backend/backend/src/books/books.controller.ts
   48:19  error  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
   74:19  error  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  116:19  error  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  143:19  error  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  171:19  error  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  198:19  error  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  227:19  error  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  247:30  error  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  257:19  error  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  275:30  error  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  286:19  error  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  305:30  error  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  315:19  error  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  371:19  error  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any

/home/scarf/repo/lib42/backend/backend/src/books/books.repository.ts
  95:21  error  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any

/home/scarf/repo/lib42/backend/backend/src/books/books.service.ts
  132:19  error  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any

/home/scarf/repo/lib42/backend/backend/src/books/likes.service.ts
  66:30  error  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any

/home/scarf/repo/lib42/backend/backend/src/histories/histories.controller.ts
  10:1   error  Definition for rule 'import/prefer-default-export' was not found  import/prefer-default-export
  22:54  error  Unexpected any. Specify a different type                          @typescript-eslint/no-explicit-any
  37:19  error  Unexpected any. Specify a different type                          @typescript-eslint/no-explicit-any

/home/scarf/repo/lib42/backend/backend/src/histories/histories.service.ts
   6:1   error  Definition for rule 'import/prefer-default-export' was not found  import/prefer-default-export
  15:20  error  Unexpected any. Specify a different type                          @typescript-eslint/no-explicit-any

/home/scarf/repo/lib42/backend/backend/src/lendings/lendings.controller.spec.ts
  11:16  error  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  19:16  error  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any

/home/scarf/repo/lib42/backend/backend/src/lendings/lendings.controller.ts
   15:30  error  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
   27:19  error  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
   62:19  error  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
   88:19  error  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  107:30  error  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  118:19  error  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any

/home/scarf/repo/lib42/backend/backend/src/middlewares/wrapAsyncController.ts
  1:1  error  Definition for rule 'import/prefer-default-export' was not found  import/prefer-default-export

/home/scarf/repo/lib42/backend/backend/src/mysql.ts
  16:63  error  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  16:84  error  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  22:7   error  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  40:11  error  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  41:12  error  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  46:9   error  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any

/home/scarf/repo/lib42/backend/backend/src/reservations/reservations.controller.spec.ts
  11:16  error  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  19:16  error  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any

/home/scarf/repo/lib42/backend/backend/src/reservations/reservations.controller.ts
   16:30  error  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
   27:19  error  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
   67:19  error  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
   81:36  error  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
   91:19  error  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  112:19  error  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  139:19  error  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any

/home/scarf/repo/lib42/backend/backend/src/reservations/reservations.service.ts
  158:15  error  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any

/home/scarf/repo/lib42/backend/backend/src/reviews/controller/reviews.controller.ts
  16:39  error  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  28:39  error  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  52:43  error  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  63:39  error  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  75:39  error  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any

/home/scarf/repo/lib42/backend/backend/src/reviews/controller/utils/errorCheck.ts
  25:20  error  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  36:20  error  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any

/home/scarf/repo/lib42/backend/backend/src/reviews/controller/utils/parseCheck.ts
   2:10  error  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  19:17  error  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  23:18  error  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any

/home/scarf/repo/lib42/backend/backend/src/reviews/service/utils/errorCheck.ts
  10:16  error  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  15:20  error  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any

/home/scarf/repo/lib42/backend/backend/src/slack/slack.controller.ts
  16:19  error  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any

/home/scarf/repo/lib42/backend/backend/src/slack/slack.service.ts
  36:20  error  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  41:71  error  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  64:37  error  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any

/home/scarf/repo/lib42/backend/backend/src/stocks/stocks.controller.ts
  20:19  error  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  35:19  error  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any

/home/scarf/repo/lib42/backend/backend/src/stocks/stocks.service.ts
  32:19  error  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any

/home/scarf/repo/lib42/backend/backend/src/tags/tags.controller.ts
   15:39  error  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
   42:39  error  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
   68:39  error  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
   80:39  error  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  134:39  error  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  168:39  error  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  200:39  error  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any

/home/scarf/repo/lib42/backend/backend/src/tags/tags.service.ts
  52:32  error  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any

/home/scarf/repo/lib42/backend/backend/src/users/users.controller.ts
   49:19  error  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
   80:19  error  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  118:19  error  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  140:39  error  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  163:19  error  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any

/home/scarf/repo/lib42/backend/backend/src/users/users.service.ts
  136:24  error  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any

/home/scarf/repo/lib42/backend/backend/src/users/users.utils.ts
  1:1  error  Definition for rule 'import/prefer-default-export' was not found  import/prefer-default-export

/home/scarf/repo/lib42/backend/backend/src/utils/parseCheck.ts
   2:10  error  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  19:17  error  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  23:18  error  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
```

</details>

--- a/backend/src/books/books.repository.ts
+++ b/backend/src/books/books.repository.ts
@@ -115,8 +115,8 @@ class BooksRepository extends Repository<Book> {
       .leftJoin(Category, 'category', 'category.id = book_info.categoryId')
       .limit(limit)
       .groupBy('book_info.id')
-      .orderBy('title', 'ASC')
       .orderBy(order, 'DESC')
+      .addOrderBy('title', 'ASC')
       .getRawMany<LendingBookList>();
     return lendingBookList;
   }

--- a/backend/src/books/books.repository.ts
+++ b/backend/src/books/books.repository.ts
@@ -88,11 +88,7 @@ class BooksRepository extends Repository<Book> {
     sort: string,
     limit: number,
   ): Promise<LendingBookList[]> {
-    const orderingArr = [
-      { createdAt: 'DESC', title: 'ASC' },
-      { lendingCnt: 'DESC', title: 'ASC' },
-    ];
-    const ordering: any = sort === 'popular' ? orderingArr[1] : orderingArr[0];
+    const order = sort === 'popular' ? 'lendingCnt' : 'createdAt';
     const lendingCondition: string = sort === 'popular'
       ? 'and lending.createdAt >= date_sub(now(), interval 42 day)'
       : '';
@@ -119,7 +115,8 @@ class BooksRepository extends Repository<Book> {
       .leftJoin(Category, 'category', 'category.id = book_info.categoryId')
       .limit(limit)
       .groupBy('book_info.id')
-      .orderBy(ordering)
+      .orderBy('title', 'ASC')
+      .orderBy(order, 'DESC')
       .getRawMany<LendingBookList>();
     return lendingBookList;
   }

--- a/backend/src/reservations/reservations.repository.ts
+++ b/backend/src/reservations/reservations.repository.ts
@@ -51,7 +51,7 @@ class ReservationsRepository extends Repository<reservation> {
   }
 
   // 유저가 연체 중인지 확인
-  async isOverdueUser(userId: number) {
+  async isOverdueUser(userId: number): Promise<boolean> {
     return this.user
       .createQueryBuilder('u')
       .select('u.id')
@@ -127,7 +127,7 @@ class ReservationsRepository extends Repository<reservation> {
     return reservedBooks;
   }
 
-  async createReservation(userId: number, bookInfoId:number) {
+  async createReservation(userId: number, bookInfoId:number): Promise<void> {
     await this.createQueryBuilder()
       .insert()
       .into(reservation)
@@ -135,7 +135,7 @@ class ReservationsRepository extends Repository<reservation> {
       .execute();
   }
 
-  async searchReservations(query: string, filter: string, page: number, limit: number) {
+  async searchReservations(query: string, filter: string, page: number, limit: number): Promise<{ meta: Meta; items: reservation[] }> {
     const searchAll = this
       .createQueryBuilder('r')
       .select('r.id', 'reservationsId')

--- a/backend/src/reservations/reservations.repository.ts
+++ b/backend/src/reservations/reservations.repository.ts
@@ -173,7 +173,7 @@ class ReservationsRepository extends Repository<reservation> {
       default:
         searchAll.andWhere({ status: 0, bookId: IsNull() });
     }
-    const items = await searchAll.offset(limit * page).limit(limit).getRawMany();
+    const items = await searchAll.offset(limit * page).limit(limit).getRawMany<reservation>();
     const totalItems = await searchAll.getCount();
     const meta : Meta = {
       totalItems,

--- a/backend/src/reservations/reservations.repository.ts
+++ b/backend/src/reservations/reservations.repository.ts
@@ -106,6 +106,7 @@ class ReservationsRepository extends Repository<reservation> {
     return lenderableBookItemNum;
   }
 
+  // Todo: return 값 수정할 것
   async alreadyLendedBooks(userId: number, bookInfoId: number) {
     const lendedBooks = this.lending
       .createQueryBuilder('l')
@@ -117,6 +118,7 @@ class ReservationsRepository extends Repository<reservation> {
     return lendedBooks;
   }
 
+  // Todo: return 값 수정할 것
   async getReservedBooks(userId: number, bookInfoId: number) {
     const reservedBooks = this
       .createQueryBuilder('r')

--- a/backend/src/reviews/repository/reviews.repository.ts
+++ b/backend/src/reviews/repository/reviews.repository.ts
@@ -42,7 +42,7 @@ export default class ReviewsRepository extends Repository<Reviews> {
     page: number,
     sort: 'ASC' | 'DESC' | undefined,
     limit: number,
-  ) {
+  ): Promise<Reviews[]> {
     const reviews = this.createQueryBuilder('reviews')
       .select('reviews.id', 'reviewsId')
       .addSelect('reviews.userId', 'reviewerId')
@@ -67,7 +67,7 @@ export default class ReviewsRepository extends Repository<Reviews> {
     }
     const ret = await reviews.offset(page * limit)
       .limit(limit)
-      .getRawMany();
+      .getRawMany<Reviews>();
     return ret;
   }
 

